mypy/build.py

```diff
@@ -1606,8 +1685,15 @@ def __init__(self,
                 # difference and just assume 'builtins' everywhere,
                 # which simplifies code.
                 file_id = '__builtin__'
-            path = find_module(file_id, manager.lib_path)
+            path = find_module(file_id, manager.lib_path, manager.options.python)
             if path:
+                if os.path.isabs(path):
```

Comment:
No, and since you misunderstood this, it definitely deserves some explanation in a comment :)

We want to silence errors in modules that are installed packages. The best heuristic that works on all Python versions that I could find is to check that if we are have a path (eg `/foo/bar/baz`) and we have a single package directory `/foo/bar`, the path  should start with `/foo/bar/` (note the last seperator). I originally formulated this differently, hence the weird code, but I think
```
if os.path.isabs(path):
    for dir in package_dirs_cache:
         # if dir is /foo and path is /foo/bar, the next character after
         # the prefix is /
         if path.startswith(dir + os.sep):
            self.ignore_all = True
```

should accomplish the same task and be cleaner. The `abspath` check is just because all of the package directories that are installed packages (and will need to be silenced by this check) will be absolute, so we can filter relative paths.